docs/components/table-view.md

```diff
@@ -0,0 +1,1039 @@
+  :headers="basicHeaders()"
+>
+  <template #tooltip-email>
+    HubSpot Id: <code>8576925e-d7e0-4ecd-8f14-15db1765e69a</code>
+  </template>
+</KTableView>
+
+```html
+<KTableView
+  :data="tableData"
+  :headers="headers"
+>
+  <template #tooltip-email>
+    HubSpot Id: <code>8576925e-d7e0-4ecd-8f14-15db1765e69a</code>
+  </template>
+</KTableView>
+```
+
+### toolbar
+
+The toolbar is rendered directly above the table and is useful for providing table controls like search or filter fields.
+
+<KTableView
+  :data="basicData"
+  :headers="basicHeaders()"
+>
+  <template #toolbar>
+    <KInput placeholder="Search">
+      <template #before>
+        <SearchIcon />
+      </template>
+    </KInput>
+    <KButton size="large">
+      <AddIcon /> Add user
+    </KButton>
+  </template>
+</KTableView>
+
+```html
+<KTableView
+  :data="tableData"
+  :headers="headers"
+>
+  <template #toolbar>
+    <KInput placeholder="Search">
+      <template #before>
+        <SearchIcon />
+      </template>
+    </KInput>
+    <KButton size="large">
+      <AddIcon /> Add user
+    </KButton>
+  </template>
+</KTableView>
+```
+
+### empty-state
+
+Slot content to be displayed when empty.
+
+### empty-state-action-icon
+
+Slot for icon to be displayed in front of action button text in empty state. See [empty state](#empty) section for example of usage of this slot.
+
+### error-state
+
+Slot content to be displayed when in error state.
+
+### action-items
+
+Slot for passing action dropdown items. See [KDropdownItem component docs](/components/dropdown#kdropdownitem) for details.
+
+Slot props:
+* `row` - table row object
+* `rowKey` - table row index
+
+:::tip NOTE
+This slot is only available when the `actions` header key is present in [`headers`](#reserved-header-keys).
+:::
+
+<KTableView
+  :data="basicData"
+  :headers="basicHeaders(true)"
+>
+  <template #action-items>
+    <KDropdownItem>
+      Edit
+    </KDropdownItem>
+    <KDropdownItem
+      danger
+      has-divider
+    >
+      Delete
+    </KDropdownItem>
+  </template>
+</KTableView>
+
+```html
+<KTableView
+  :data="tableData"
+  :headers="headers"
+>
+  <template #action-items>
+    <KDropdownItem>
+      Edit
+    </KDropdownItem>
+    <KDropdownItem
+      danger
+      has-divider
+    >
+      Delete
+    </KDropdownItem>
+  </template>
+</KTableView>
+```
+
+### after
+
+Slot for content that should be rendered below the table. For example, could be used for displaying custom pagination component.
+
+<KTableView
+  :data="basicData"
+  :headers="basicHeaders()"
+>
+  <template #after>
+    <KPagination
+      :items="basicData"
+      :total-count="basicData.length"
+    />
+  </template>
+</KTableView>
+
+```html
+<KTableView
+  :data="tableData"
+  :headers="headers"
+>
+  <template #after>
+    <KPagination
+      :items="tableData"
+      :total-count="tableData.length"
+      @page-change="onPageChange"
+    />
+  </template>
+</KTableView>
+```
+
+## Events
+
+### Row Events
```

Comment:
issue: the row events and cell events examples are pretty confusing; there is so much going on in each cell I'm not sure what the example is trying to show me. Can we simplify?

Reply:
I mean they are but I think it's one of those things that you need to know what you're looking for? The button, the input field and an external link are there to just demonstrate that you can both click on row/cell and other interactive elements within the table and handle those clicks separately.

Reply:
In other words, it doesn't look confusing once you know what too look for in that example but I also don't know how to make it less confusing for those who don't know what exactly they are looking for

Reply:
I don't think the "rowEvents" and "cellEvents" examples should look exactly the same - I can't tell what the difference is between the two examples

![image](https://github.com/user-attachments/assets/ac4d5203-0396-48d5-93d9-91a9aa2145f1)

We should rework this

Reply:
Oh, that's a good point, they don't have to be identical. Changed them up a bit and added a sentence in front of the example to better clarify what the example is demonstrating.